### PR TITLE
Turn IncomingStreamHandler into FramedConnection.Listener.

### PR DESCRIPTION
**Pull request by swankjesse**
Add a callback invoked on settings changes. The concurrency here
is a little awkward because the calls into the listener are not
serialized.

This is going to be used in a follow up change to keep the
connection's allocation limit in sync.

### CODE REVIEW

**Comment by JakeWharton** [comment]
```
okhttp/src/main/java/com/squareup/okhttp/internal/framed/FramedConnection.java:915:17: 'abstract' modifier out of order with the JLS suggestions.
```

**Comment by swankjesse** [comment]
Always forget static abstract or abstract static!

File: okhttp/src/main/java/com/squareup/okhttp/internal/framed/FramedConnection.java [hunk 1]
```diff
@@ -905,4 +910,34 @@ private void pushResetLater(final int streamId, final ErrorCode errorCode) {
...
+  public static abstract class Listener {
+    public static final Listener REFUSE_INCOMING_STREAMS = new Listener() {
+      @Override public void onStream(FramedStream stream) throws IOException {
+        stream.close(ErrorCode.REFUSED_STREAM);
+      }
+    };
+
+    /**
+     * Handle a new stream from this connection's peer. Implementations should
+     * respond by either {@link FramedStream#reply replying to the stream} or
```
Comment (JakeWharton) [inline okhttp/src/main/java/com/squareup/okhttp/internal/framed/FramedConnection.java]: linkplain

File: okhttp/src/main/java/com/squareup/okhttp/internal/framed/FramedConnection.java [hunk 2]
```diff
@@ -905,4 +910,34 @@ private void pushResetLater(final int streamId, final ErrorCode errorCode) {
...
+     * {@link FramedStream#close closing it}. This response does not need to be
```
Comment (JakeWharton) [inline okhttp/src/main/java/com/squareup/okhttp/internal/framed/FramedConnection.java]: linkplain

File: okhttp-tests/src/test/java/com/squareup/okhttp/internal/framed/Spdy3ConnectionTest.java [hunk 3]
```diff
@@ -150,8 +150,8 @@
 
     // play it back
     final AtomicInteger receiveCount = new AtomicInteger();
-    IncomingStreamHandler handler = new IncomingStreamHandler() {
-      @Override public void receive(FramedStream stream) throws IOException {
+    FramedConnection.Listener handler = new FramedConnection.Listener() {
```
Comment (JakeWharton) [inline okhttp-tests/src/test/java/com/squareup/okhttp/internal/framed/Spdy3ConnectionTest.java]: nit: could be renamed `listener`

File: okhttp-tests/src/test/java/com/squareup/okhttp/internal/framed/Spdy3ConnectionTest.java [hunk 4]
```diff
@@ -181,14 +181,14 @@
...
```
Comment (JakeWharton) [inline okhttp-tests/src/test/java/com/squareup/okhttp/internal/framed/Spdy3ConnectionTest.java]: nit: could be renamed `listener`

File: okhttp-tests/src/test/java/com/squareup/okhttp/internal/framed/Spdy3ConnectionTest.java [hunk 5]
```diff
@@ -616,8 +628,8 @@
...
```
Comment (JakeWharton) [inline okhttp-tests/src/test/java/com/squareup/okhttp/internal/framed/Spdy3ConnectionTest.java]: nit: could be renamed `listener`

**Comment by JakeWharton** [comment]
LGTM

File: okhttp-tests/src/test/java/com/squareup/okhttp/internal/framed/Spdy3ConnectionTest.java (hunk 5 above)
Comment (swankjesse) [inline okhttp-tests/src/test/java/com/squareup/okhttp/internal/framed/Spdy3ConnectionTest.java]: Fixed!

File: okhttp-tests/src/test/java/com/squareup/okhttp/internal/framed/Spdy3ConnectionTest.java (hunk 4 above)
Comment (swankjesse) [inline okhttp-tests/src/test/java/com/squareup/okhttp/internal/framed/Spdy3ConnectionTest.java]: Fixed!

File: okhttp-tests/src/test/java/com/squareup/okhttp/internal/framed/Spdy3ConnectionTest.java (hunk 3 above)
Comment (swankjesse) [inline okhttp-tests/src/test/java/com/squareup/okhttp/internal/framed/Spdy3ConnectionTest.java]: Fixed!

File: okhttp/src/main/java/com/squareup/okhttp/internal/framed/FramedConnection.java (hunk 2 above)
Comment (swankjesse) [inline okhttp/src/main/java/com/squareup/okhttp/internal/framed/FramedConnection.java]: Fixed!

File: okhttp/src/main/java/com/squareup/okhttp/internal/framed/FramedConnection.java (hunk 1 above)
Comment (swankjesse) [inline okhttp/src/main/java/com/squareup/okhttp/internal/framed/FramedConnection.java]: Fixed!

**Pull request merged**